src/typo-tolerance/setup/setupMeili.js

```diff
@@ -0,0 +1,66 @@
+const { MeiliSearch } = require('meilisearch')
+const dataset = require('./books.json')
+const { watchTasks, populateIndex } = require('./setupFunctions')
+require('dotenv').config()
+
+const DEFAULT_INDEX = 'books_default'
+const TYPO_TOLERANT_INDEX = 'books_typo'
+
+const settings = {
+  distinctAttribute: null,
+  searchableAttributes: [
+    'title',
+    'isbn',
+    'authors',
+    'shortDescription'
+  ],
+  displayedAttributes: [
+    'title',
+    'isbn',
+    'authors',
+    'shortDescription',
+    'thumbnailUrl'
+  ]
+}
+
+const defaultTypoTolerance = {
+  minWordSizeForTypos: {
+    oneTypo: 5,
+    twoTypos: 9
+  }
+}
+const customTypoTolerance = {
+  disableOnAttributes: ['isbn'],
+  minWordSizeForTypos: {
+    oneTypo: 2,
+    twoTypos: 4
+  }
+}
+
+;(async () => {
+  // Create client
+  const client = new MeiliSearch({
+    host: process.env.VITE_MEILISEARCH_HOST,
+    apiKey: process.env.VITE_MEILISEARCH_API_KEY
+  })
+
+  const indexArray = [
+    { uid: DEFAULT_INDEX, typoTolerance: defaultTypoTolerance },
+    { uid: TYPO_TOLERANT_INDEX, typoTolerance: customTypoTolerance }
+  ]
+  // Create index and add settings and documents
+  await Promise.all(
+    indexArray.map(async index => {
+      await client.createIndex(index.uid)
+      await populateIndex(settings, index, dataset, client)
+    })
+  )
```

Comment:
Create index is not mandatory anymore as `addDocuments` creates the index

```suggestion
  await Promise.all(
    indexArray.map(async index => {
      await populateIndex(settings, index, dataset, client)
    })
  )
```

Reply:
Will this modification work since populateIndex adds the settings first with `await client.index(uid).updateSettings({ ...settings, typoTolerance })`?

Reply:
It does `updateSettings` creates the index!